File: 011_container_with_most_water/Solution.py
class Solution:
    def maxArea(self, height: list) -> int:
        if not height or len(height) == 0: return 0

        l, r = 0, len(height) - 1
        ans, area = 0, 0

        while l < r:
            area = min(height[l], height[r]) * (r - l)
            ans = max(area, ans)
            if height[l] < height[r]: l += 1
            else: r -= 1

        return ans


if __name__ == '__main__':
    solution = Solution()
    print(solution.maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7]))

File: 011_container_with_most_water/Solution.java
public class Solution {
    public int maxArea(int[] height) {
        if (height == null || height.length == 0) return 0;

        int arrLen = height.length;
        int l = 0, r = arrLen - 1;
        int ans = 0, area = 0;

        while (l < r) {
            area = Math.min(height[l], height[r]) * (r - l);
            ans = Math.max(ans, area);
            if (height[l] > height[r]) r--;
            else l++;
        }

        return ans;
    }

    public static void main(String[] args) {
        int[] heightArr = new int[]{1, 8, 6, 2, 5, 4, 8, 3, 7};
        Solution solution = new Solution();
        System.out.println(solution.maxArea(heightArr));      
    }
}
